PES-1379 Add php code sniffer

Mrakor opened the pull request.

Mrakor committed:
PES-1379 Add support for phpcs

add composer.json

Mrakor committed:
PES-1379 Add original OC3 files for phpcs

Mrakor committed:
PES-1379 Fix original OC3 files

Mrakor committed:
PES-1379 phpcs ruleset and Packetery sniffs

Mrakor committed:
PES-1379 Disallow tabs for indents

fix example for 1 file check

Mrakor committed:
PES-1379 fix a typo

Mrakor committed:
PES-1379 phpcs - less strict rules for phpdoc comments

Mrakor committed:
PES-1379 add rule to sniff for duplicate spaces

add filter to skip checking of model classes, which are not zailkovna*.php

Mrakor committed:
PES-1379 CS fix of filter

packeta-user commented on phpcs.xml:
```diff
@@ -0,0 +1,93 @@
+    <arg name="report" value="full"/>
+    <arg name="colors"/>
+    <arg value="sp"/>
+    <arg name="filter" value="./phpcs/Packetery/Filters/ZasilkovnaModels.php"/>
```
Nelze toho samého docílit použitím "exclude-pattern" ?

packeta-user commented on phpcs.xml:
```diff
@@ -0,0 +1,93 @@
+    <file>./upload/catalog/model/extension/shipping/zasilkovna.php</file>
+
+    <!-- language files to be tested -->
+    <file>./upload/catalog/language/en-gb/extension/shipping/zasilkovna.php</file>
```
Nejde v rámci elementu "file" také použít regulární výraz, který by zajistil, že místo en-gb bude jakýkoli název adresáře ? Tím bychom měli vyřešenou situaci, pokud do modulu přidáme další jazykový soubor.

packeta-user commented on phpcs.xml:
```diff
@@ -0,0 +1,93 @@
+    <file>./phpcs/Packetery</file>
+
+<!-- For testing: comment up the above and uncomment the below or use your own php file-->
+<!--    <file>./upload/system/library/Packetery/Order/OrderRepository.php</file>-->
```
K čemu je toto dobré ? Nepoužívá se pro tyto účely spíš to že phpcs na konkrétní soubor, jehož cestu uvedu v terminálu za phpcs ?

Mrakor committed:
PES-1379 CR fixes

use exclude-pattern instead of filter
use exclude-pattern for language files,so future translations are auto-included
remove unnecessary comment

Mrakor committed:
PES-1379 add phpcs/vendor to .gitignore
